spec/serializers/rest/account_relationship_severance_event_serializer_spec.rb

```diff
@@ -5,7 +5,7 @@
 RSpec.describe REST::AccountRelationshipSeveranceEventSerializer do
   subject { serialized_record_json(record, described_class) }
 
-  let(:record) { Fabricate.build :account_relationship_severance_event, id: 123 }
+  let(:record) { Fabricate.build :account_relationship_severance_event, id: 123, created_at: DateTime.new(2024, 11, 28, 16, 20, 0) }
```

Comment:
```suggestion
  let(:record) { Fabricate.build :account_relationship_severance_event, id: 123, created_at: DateTime.new(2024, 11, 28, 16, 20, 0).utc }
```

Reply:
Can you help me understand the rationale for that change?

As long as it's a serialized as an RFC3339 compatible date client's shouldn't mind what the timezone designator is. Even if https://docs.joinmastodon.org/api/datetime-format/ was updated to say the server will only ever return dates in UTC (so the timezone designator would always be `Z`) clients should be prepared for it not to be.

Also, I suspect it provides more opportunities for bugs where the server inadvertently returns a correctly formatted RFC3339 datetime with a timezone offset.

So I'm concerned that being explicit that it must be UTC in the tests is overspecifying the test relative to the real-world behaviour that's expected.

Reply:
Where did you see the spec failure?

If I run `rspec spec/serializers/rest/account_relationship_severance_event_serializer_spec.rb` in a clean client I get:

```
Run options: exclude {:streaming=>true, :search=>true, :js=>true}

Randomized with seed 19649
..

Finished in 1.98 seconds (files took 8.68 seconds to load)
2 examples, 0 failures

Randomized with seed 19649

Coverage report generated for RSpec to /home/nik/projects/mastodon/mastodon/coverage.
Line Coverage: 4.44% (1830 / 41211)
Branch Coverage: 3.77% (49 / 1300)
```

> The application code uses, or at least should use, UTC dates everywhere.

Two things here:

1. If I've understood Rails correctly (and I may not have), everything should be using whatever timezone Rails is configured to use (the default is UTC). So if you need to put `.utc` here to get a test to pass that suggests something is wrong somewhere else.

2. I don't think that should be relevant here. Specifically, I think that should be treated as an internal Mastodon decision. For the purposes of conformance and client interoperability (which is what the serializer specs test) it is sufficient that this is an RFC3339 datetime. Whether the timezone designator is `Z` or an offset like `+02:00` shouldn't matter. And  "Mastodon will only ever emit RFC3339 datetimes in UTC" seems like an unnecessarily strong external commitment to make.